crates/bevy_reflect/src/type_name.rs

```diff
@@ -0,0 +1,77 @@
+use std::borrow::Cow;
+
+/// Provide the name of the type as string.
+pub trait TypeName {
+    fn name() -> Cow<'static, str>;
+}
+
+/// A object-safe version of [`TypeName`].
+/// Automatically implemented via blanket implementation.
+pub trait ReflectTypeName {
+    fn type_name(&self) -> Cow<str>;
```

Comment:
Why is this not `Cow<'static, str>`?

Reply:
Because `Dynamic*` struct (i.e. `DynamicArray`, `DynamicTuple`, etc.) have a field `name` with the name of the type.
So the lifetime must be bound to `&self` to return a ref to this field.